src/Http/Http/src/Internal/StreamCopyOperation.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Http
+{
+    // FYI: In most cases the source will be a FileStream and the destination will be to the network.
+    internal static class StreamCopyOperation
```

Comment:
I wonder if it still makes sense to use this. The CopyToAsync operation has been optimized in .NET Core 2.x and I'm not sure this still has the same benefits.

Reply:
This one wasn't for performance, it was to allow you to only copy a specific range.

Reply:
It was for copying using pooled memory. If it was for copying a specific range, it would first call seek then delegate to CopyToAsync

Reply:
CopyToAsync can't limit the length of the range copied, it always copies the remainder of the stream.

Reply:
Ah good point. Can you add a comment on this type for that behavior.

Reply:
I'd think the method signature covered that. CopyToAsync has not Length parameter.

What I really need to do here is deduplicate the code with the copy in Http.Extensions.

Reply:
Add a comment so that future developers don't need to ask and no the method signature doesn't cover it.